### PR TITLE
feat: port `@u22n/storage` to Hono

### PR DESCRIPTION
**Pull request by BlankParticle**
## What does this PR do?

Moves the storage app to Hono from Nitro.

Needs manual testing of all endpoints on staging server.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Required

- [ ] Read [Contributing Guide](https://github.com/un/inbox/blob/main/CONTRIBUTING.md)
- [ ] Self-reviewed my own code
- [ ] Tested my code in a local environment
- [ ] Commented on my code in hard-to-understand areas
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the UnInbox Docs if changes were necessary

### CODE REVIEW

**Comment by BlankParticle** [comment]
* **#363** <a href="https://app.graphite.dev/github/pr/un/inbox/363?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @BlankParticle and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/storage/src/app.ts [hunk 1]
```diff
@@ -0,0 +1,565 @@
...
+  }
+  // Copy response as fetch headers are immutable
+  return new Response(res.body, res);
+});
+
+app
+  // Health check endpoint
+  .get('/', (c) => c.json({ status: "I'm Alive 🏝️" }))
+  // Attachment download endpoint
+  .get(
```
Comment (McPizza0) [inline apps/storage/src/app.ts:115]: why is this chained?
I would rather it was app.get per route to make it easier to skim through

File: apps/storage/src/app.ts [hunk 2]
```diff
@@ -0,0 +1,565 @@
...
+          { error: 'Error while fetching attachment' },
+          { status: 500 }
+        );
+      }
+      // Copy response as fetch headers are immutable
+      return new Response(res.body, res);
+    }
+  )
+  // Avatar upload endpoint
+  .post(
```
Comment (McPizza0) [inline apps/storage/src/app.ts:198]: same here
lets not chain
makes it easier to parse and move things around later

File: apps/storage/src/app.ts [hunk 3]
```diff
@@ -0,0 +1,565 @@
...
+            .where(eq(teams.publicId, publicId as TypeId<'teams'>));
+          break;
+        }
+      }
+
+      return c.json({ avatarTimestamp });
+    }
+  )
+  // Attachment presign endpoint
+  .get(
```
Comment (McPizza0) [inline apps/storage/src/app.ts:420]: same

File: apps/storage/src/app.ts [hunk 4]
```diff
@@ -0,0 +1,565 @@
...
+        orgPublicId: orgPublicId,
+        publicId: attachmentPublicId,
+        filename: filename
+      });
+
+      return c.json({ publicId: attachmentPublicId, signedUrl });
+    }
+  )
+  // Attachment internal presign endpoint
+  .post(
```
Comment (McPizza0) [inline apps/storage/src/app.ts:481]: s

**Review by McPizza0 — CHANGES_REQUESTED** [review]

**Review by McPizza0 — APPROVED** [review]

**Pull request merged**